jetty-core/jetty-websocket/jetty-websocket-core-common/src/main/java/org/eclipse/jetty/websocket/core/messages/DispatchedMessageSink.java

```diff
@@ -94,74 +94,70 @@
  */
 public abstract class DispatchedMessageSink extends AbstractMessageSink
 {
-    private CompletableFuture<Void> dispatchComplete;
-    private MessageSink typeSink;
     private final Executor executor;
+    private volatile CompletableFuture<Void> dispatchComplete;
+    private MessageSink typeSink;
 
-    public DispatchedMessageSink(CoreSession session, MethodHandle methodHandle)
+    public DispatchedMessageSink(CoreSession session, MethodHandle methodHandle, boolean autoDemand)
```

Comment:
If `autoDemand` is false, then what demand(1) mean?  1 frame or 1 message?

At the very least javadoc is needed.

Reply:
This is WebSocket core, so it only deals with frames.

I have expanded and fixed the javadocs for `AbstractMessageSink` and `DispatchedMessageSink`.